qcschema/dev/QCSchema.schema

```diff
@@ -0,0 +1,1575 @@
+            "description": "The MolSSI Quantum Chemistry Schema\n\nParameters\n----------\nid : str, Optional\n    The optional ID for the computation.\nschema_name : ConstrainedStrValue, Default: qcschema_input\n    The QCSchema specification this model conforms to. Explicitly fixed as qcschema_input.\nschema_version : int, Default: 1\n    The version number of ``schema_name`` to which this model conforms.\nmolecule : :class:`Molecule`\n    The molecule to use in the computation.\ndriver : {energy,gradient,hessian,properties}\n    Allowed computation driver values.\nmodel : :class:`Model`\n    The computational molecular sciences model to run.\nkeywords : Dict[str, Any], Default: {}\n    The program-specific keywords to be used.\nprotocols : :class:`AtomicResultProtocols`, Optional\n    Protocols regarding the manipulation of computational result data.\nextras : Dict[str, Any], Default: {}\n    Additional information to bundle with the computation. Use for schema development and scratch space.\nprovenance : :class:`Provenance`, Optional\n    Provenance information.",
+            "type": "object",
+            "properties": {
+                "id": {
+                    "title": "Id",
+                    "description": "The optional ID for the computation.",
+                    "type": "string"
+                },
+                "schema_name": {
+                    "title": "Schema Name",
+                    "description": "The QCSchema specification this model conforms to. Explicitly fixed as qcschema_input.",
+                    "default": "qcschema_input",
+                    "pattern": "^(qc_?schema_input)$",
+                    "type": "string"
+                },
+                "schema_version": {
+                    "title": "Schema Version",
+                    "description": "The version number of ``schema_name`` to which this model conforms.",
+                    "default": 1,
+                    "type": "integer"
+                },
+                "molecule": {
+                    "title": "Molecule",
+                    "description": "The molecule to use in the computation.",
+                    "allOf": [
+                        {
+                            "$ref": "#/definitions/Molecule"
+                        }
+                    ]
+                },
+                "driver": {
+                    "$ref": "#/definitions/DriverEnum"
+                },
+                "model": {
+                    "title": "Model",
+                    "description": "The computational molecular sciences model to run.",
+                    "allOf": [
+                        {
+                            "$ref": "#/definitions/Model"
+                        }
+                    ]
+                },
+                "keywords": {
+                    "title": "Keywords",
+                    "description": "The program-specific keywords to be used.",
+                    "default": {},
```

Comment:
If the notion of a separate list of required fields for full instances catches on, this and certain other fields need to have empty dictionary default so that access is easy without a bunch of error checking.